### PR TITLE
Update use cases for 'local' packages

### PR DESCRIPTION
**Pull request by jtcohen6**
Context: https://github.com/dbt-labs/dbt-core/issues/8541#issuecomment-1706982106

## What are you changing in this pull request and why?

There are multiple legitimate use cases for 'local' packages, beyond the one that's currently documented — especially as we see more multi-project setups.

### CODE REVIEW

File: website/docs/docs/build/packages.md [hunk 1]
```diff
@@ -284,18 +284,35 @@ packages:
...
+```yaml
+packages:
+  # not recommended - these support for these patterns vary
+  - local: /../../redshift   # relative path to a parent directory
+  - local: /opt/dbt/redshift # absolute path on the system
+```
+
+</File>
+
+As such, there are a few specific use cases where a "local" package is recommended:
```
Comment (mirnawong1) [inline website/docs/docs/build/packages.md]:
```suggestion
There are a few specific use cases where we recommend using a "local" package:
```

File: website/docs/docs/build/packages.md [hunk 2]
```diff
@@ -284,18 +284,35 @@ packages:
...
+1. A monorepo containing multiple projects, each nested in a subdirectory. "Local" packages enable combining projects together, for coordinated development and deployment.
```
Comment (mirnawong1) [inline website/docs/docs/build/packages.md]: sugg having some bolded categories so it's pretty clear from the get go

```suggestion
1. **Monorepo** &mdash; When you have multiple projects, each nested in a subdirectory, within a monorepo. "Local" packages allow you to combine projects for coordinated development and deployment.
```

File: website/docs/docs/build/packages.md [hunk 3]
```diff
@@ -284,18 +284,35 @@ packages:
...
+2. Testing changes to one project/package within the context of a downstream project/package that uses it. By temporarily switching the installation to a "local" package, you can make changes to the former and immediately test them in the latter, enabling quicker iteration. This is similar to [editable installs](https://pip.pypa.io/en/stable/topics/local-project-installs/) in Python.
```
Comment (mirnawong1) [inline website/docs/docs/build/packages.md]:
```suggestion
2. **Testing changes** &mdash; To test changes in one project or package within the context of a downstream project or package that uses it. By temporarily switching the installation to a "local" package, you can make changes to the former and immediately test them in the latter for quicker iteration. This is similar to [editable installs](https://pip.pypa.io/en/stable/topics/local-project-installs/) in Python.
```

File: website/docs/docs/build/packages.md [hunk 4]
```diff
@@ -284,18 +284,35 @@ packages:
...
+3. A nested project that defines fixtures and tests for a project of utility macros — for example, [the integration tests within the `dbt-utils` package](https://github.com/dbt-labs/dbt-utils/tree/main/integration_tests)
```
Comment (mirnawong1) [inline website/docs/docs/build/packages.md]:
```suggestion
3. **Nested project** &mdash; When you have a nested project that defines fixtures and tests for a project of utility macros, like [the integration tests within the `dbt-utils` package](https://github.com/dbt-labs/dbt-utils/tree/main/integration_tests).
```

File: website/docs/docs/build/packages.md [hunk 5]
```diff
@@ -284,18 +284,35 @@ packages:
 </File>
 
 ### Local packages
-Packages that you have stored locally can be installed by specifying the path to the project, like so:
+A "local" package is another dbt project that you have access to from within the local file system. It can be installed by specifying the path to the project. The best-supported pattern is when the project is nested within a subdirectory, relative to the current project's directory.
```
Comment (mirnawong1) [inline website/docs/docs/build/packages.md]:
```suggestion
A "local" package is a dbt project accessible from your local file system. You can install it by specifying the project's path. It works best when you nest the project within a subdirectory relative to your current project's directory.
```

File: website/docs/docs/build/packages.md [hunk 6]
```diff
@@ -284,18 +284,35 @@ packages:
...
 </File>
 
-Local packages should only be used for specific situations, for example, when testing local changes to a package.
+Other patterns may work in some cases, but not always. For example, if you install this project as a package elsewhere, or try running it on a different system, the relative and absolute paths will yield the same results.
+
+<File name='packages.yml'>
+
+```yaml
+packages:
+  # not recommended - these support for these patterns vary
```
Comment (mirnawong1) [inline website/docs/docs/build/packages.md]: did you mean 'the'?
```suggestion
  # not recommended - the support for these patterns vary
```

**Comment by mirnawong1** [comment]
hey @jtcohen6 , thanks for opening this up and clarifying! I've left some suggestions for clarity and readability - if you're happy with them, i can go ahead and commit/merge

**Comment by jtcohen6** [comment]
Thank you @mirnawong1! I took a quick look over your suggestions - I'm happy with all of them!

**Comment by mirnawong1** [comment]
thanks so much for the quick response @jtcohen6 , I'll get these committed + merged for you ✨

File: website/docs/docs/build/packages.md [hunk 7]
```diff
@@ -284,18 +284,35 @@ packages:
...
 </File>
 
-Local packages should only be used for specific situations, for example, when testing local changes to a package.
+Other patterns may work in some cases, but not always. For example, if you install this project as a package elsewhere, or try running it on a different system, the relative and absolute paths will yield the same results.
+
+<File name='packages.yml'>
+
+```yaml
+packages:
+  # not recommended - the support for these patterns vary
```
Comment (mirnawong1) [inline website/docs/docs/build/packages.md]:
```suggestion
  # not recommended - support for these patterns vary
```

**Review by mirnawong1 — APPROVED** [review]
🚢

**Comment by mirnawong1** [comment]
Screenshot 2023-09-08 at 13 33 11

**Pull request merged**